### DISCUSSION TITLE
Unable to Run code on VS C++

### DISCUSSION THREAD

**Discussion by chokostica** [community/community#66380] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

[csma-cd-protocol-main.zip](https://github.com/community/community/files/12551399/csma-cd-protocol-main.zip)

I cloned the attached code from github into Visual Studio C++. I am unable to run it. It appears there is no .sln file. How do I run this code successfully on VS C++?

**Comment by reeveng** [comment]
Maybe you can find a solution on the new for you page?

**Reply by chokostica** [reply]
Hello reeveng, I didn't understand where you said I can find the solution. Please clarify.

Here is the repo where I downloaded the code:
https://github.com/Bilal-Mhmd/csma-cd-protocol

**Reply by reeveng** [reply]
https://github.com , they recently added a new homepage, maybe you can find the answer on there? it is supposed to be an all knowing algorithm that helps you :sunglasses:

**Reply by chokostica** [reply]
Not clear... Thank you.

**Reply by reeveng** [reply]
Previously was me protesting a recent change, on a serious note:
If there are no project files like `.sln`, you may need to create a new project in Visual Studio and then add the existing source code files to it.
`.sln` is VS specific, the people who created the repository probably didn't use VS.

**Reply by chokostica** [reply]
Many thanks for this...

But I am new to it. I can create a new VS project with the .sln. However, the project I downloaded has many files, do I copy each and past in the same .sln file? Which ones comes first, and which follows? In what sequence.

Apologies I am newbie. I am asking too many questions, cos I want to learn.

**Reply by reeveng** [reply]
No worries man, I don't think the order matters too much, and you can probably import them all at once.
There is most likely even an "import existing project" which you could try if copying them all fails.
I myself however, haven't really used Visual Studio yet, but most IDE's operate pretty similarly.

[This stack-overflow](https://stackoverflow.com/questions/4853550/import-existing-c-source-code-into-visual-studio) could come in handy.

**Reply by chokostica** [reply]
Thank Man. I will try it and revert.

**Reply by reeveng** [reply]
Lemme know how it goes, me/someone else might be able to further assist you if needed :wink:

**Reply by chokostica** [reply]
Hello Man... I have created an .sln project, Hello world, and it works fine.

I tried adding my CSMA/CD project I attached before on this discussion (downloaded from Github), it seems not adding to the Hello world project.

I dunno what to do next.

**Reply by HexTTX** [reply]
你好、我的朋友，你这个问题应该尝试从Visual Studio中创建远程仓库，我向来都是如此。
